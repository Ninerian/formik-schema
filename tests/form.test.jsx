import { Form } from '../lib/formik-schema';
import renderer from 'react-test-renderer';
import React from 'react';


const testSchema2FormSnapshot = (testTitle, schema) => test(testTitle, () => {
  const component = renderer.create(
    <Form
      schema={schema}
      onSubmit={() => {}}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

testSchema2FormSnapshot('Example from https://getbootstrap.com/docs/4.0/components/forms/#horizontal-form',
  {
    fields: [
      {
        name: "email",
        title: "Email",
        type: "email",
      },
      {
        name: "password",
        title: "Mot de passe",
        type: "password",
      },
      {
        name: "accept",
        title: "I accept.",
        type: "checkbox",
        description: "Common, accept it!",
      },
    ],
  }
);

testSchema2FormSnapshot('Form with a choice and a default',
  {
    fields: [
      {
        name: "dog_or_cat",
        title: "Are you rather a dog or cat person ?",
        type: "choices",
        options: [
          {
            value: "cat",
            title: "Chat",
          },
          {
            value: "dog",
            title: "Chien",
            selected: true,
          },
        ],
      },
    ],
  }
);
