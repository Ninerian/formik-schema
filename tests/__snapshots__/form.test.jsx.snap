// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example from https://getbootstrap.com/docs/4.0/components/forms/#horizontal-form 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group row"
  >
    <label
      className="col-3 col-form-label"
      htmlFor="email"
    >
      Email
    </label>
    <div
      className="col"
    >
      <input
        className="form-control"
        id="email"
        name="email"
        onChange={[Function]}
        type="email"
        value={undefined}
      />
    </div>
  </div>
  <div
    className="form-group row"
  >
    <label
      className="col-3 col-form-label"
      htmlFor="password"
    >
      Mot de passe
    </label>
    <div
      className="col"
    >
      <input
        className="form-control"
        id="password"
        name="password"
        onChange={[Function]}
        type="password"
        value={undefined}
      />
    </div>
  </div>
  <div
    className="form-group row"
  >
    <label
      className="col-3 col-form-label"
      htmlFor="accept"
    >
      I accept.
    </label>
    <div
      className="col"
    >
      <div
        className="form-check"
      >
        <label
          className="form-check-label"
        >
          <input
            checked={undefined}
            className="form-check-input"
            name="accept"
            onChange={[Function]}
            type="checkbox"
          />
           
          Common, accept it!
        </label>
      </div>
    </div>
  </div>
  <button
    className="btn btn-primary"
    type="submit"
  >
    OK
  </button>
</form>
`;

exports[`Form with a choice and a default 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group row"
  >
    <label
      className="col-3 col-form-label"
      htmlFor="dog_or_cat"
    >
      Are you rather a dog or cat person ?
    </label>
    <div
      className="col"
    >
      <div
        className="form-check"
      >
        <label
          className="form-check-label"
        >
          <input
            checked={false}
            className="form-check-input"
            id="dog_or_cat_cat"
            name="dog_or_cat"
            onChange={[Function]}
            type="radio"
            value="cat"
          />
           
          Chat
        </label>
      </div>
      <div
        className="form-check"
      >
        <label
          className="form-check-label"
        >
          <input
            checked={false}
            className="form-check-input"
            id="dog_or_cat_dog"
            name="dog_or_cat"
            onChange={[Function]}
            type="radio"
            value="dog"
          />
           
          Chien
        </label>
      </div>
    </div>
  </div>
  <button
    className="btn btn-primary"
    type="submit"
  >
    OK
  </button>
</form>
`;
